d2game/d2player/game_controls.go

```diff
@@ -645,9 +647,31 @@ func (g *GameControls) onCloseHeroStatsPanel() {
 	g.updateLayout()
 }
 
+func (g *GameControls) toggleLeftSkillPanel() {
+	if !g.HelpOverlay.IsOpen() {
+		g.inventory.Close()
+		g.skilltree.Close()
+		g.questLog.Close()
+		g.heroStatsPanel.Close()
+		g.hud.skillSelectMenu.ToggleLeftPanel()
+	}
+}
+
+func (g *GameControls) toggleRightSkillPanel() {
+	if !g.HelpOverlay.IsOpen() {
+		g.inventory.Close()
+		g.skilltree.Close()
+		g.questLog.Close()
+		g.heroStatsPanel.Close()
+		g.hud.skillSelectMenu.ToggleRightPanel()
+	}
+}
+
```

Comment:
For now this is fine, but I don't like how the logic is hard-coded in this file. The operation of showing/hiding panels should be _way_ more generic (#965).

When we add the charsi-imbue, or the Larzuk-socketing panels we also need to update all of the panel open/close logic...? Seems wrong, no?

Reply:
Yeah, I completely agree with you. I just want to point, what should be here included.